Emitter/Emitter.Subscribe.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Emitter.Messages;
+using Emitter.Utility;
+
+namespace Emitter
+{
+    public partial class Connection
+    {
+        #region Subscribe
+
+        /// <summary>
+        /// Asynchronously subscribes to a particular channel of emitter.io service. Uses the default
+        /// key that should be specified in the constructor.
+        /// </summary>
+        /// <param name="channel">The channel to subscribe to.</param>
+        /// <param name="handler">The callback to be invoked every time the message is received.</param>
+        /// <returns>The message identifier for this operation.</returns>
+        [Obsolete("The On method is obsolete, please use the Subscribe method instead.")]
+        public ushort On(string channel, MessageHandler handler)
+        {
+            return this.Subscribe(channel, handler);
+        }
+
+        /// <summary>
+        /// Asynchronously subscribes to a particular channel of emitter.io service.
+        /// </summary>
+        /// <param name="key">The key to use for this subscription request.</param>
+        /// <param name="channel">The channel to subscribe to.</param>
+        /// <param name="handler">The callback to be invoked every time the message is received.</param>
+        /// <returns>The message identifier for this operation.</returns>
+        [Obsolete("The On method is obsolete, please use the Subscribe method instead.")]
+        public ushort On(string key, string channel, MessageHandler handler)
+        {
+            return this.Subscribe(key, channel, handler);
+        }
+
+        /// <summary>
+        /// Asynchronously subscribes to a particular channel of emitter.io service.
+        /// </summary>
+        /// <param name="key">The key to use for this subscription request.</param>
+        /// <param name="channel">The channel to subscribe to.</param>
+        /// <param name="handler">The callback to be invoked every time the message is received.</param>
+        /// <param name="last">The last x messages to request when we subcribe.</param>
+        /// <returns>The message identifier for this operation.</returns>
+        [Obsolete("The On method is obsolete, please use the Subscribe method with the WithLast option instead.")]
+        public ushort On(string key, string channel, MessageHandler handler, int last)
```

Comment:
Let's bump the major version and actually delete the code.

Reply:
See new PR